File: Ejercicio2/FactoriaMontaña.py
from .FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
from .Carrera import Carrera
from .Bicicleta import Bicicleta
from .CarreraMontaña import CarreraMontaña
from .BicicletaMontaña import BicicletaMontaña
from typing import List

class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def crear_bicicleta(self) -> Bicicleta:
        return BicicletaMontaña()

    def crear_carrera(self, bicis: List[Bicicleta]) -> Carrera:
        return CarreraMontaña(bicis)

File: P1/Ejercicio1/FactoriaMontaña.java
package Ejercicio1;

import java.util.ArrayList;

public class FactoriaMontaña implements FactoriaCarreraYBicicleta {

    @Override
    public Bicicleta crearBicicleta() {
        return new BicicletaMontaña();
    }

    @Override
    public Carrera crearCarrera(ArrayList<Bicicleta> bicis) {
        return new CarreraMontaña(bicis);
    }

}
